src/name_bazaar/server/emailer/templates.cljs

```diff
@@ -5,17 +5,24 @@
             [goog.string :as gstring]
             [name-bazaar.shared.constants :as shared-constants]))
 
-(defn- form-link [offering]
-  (str "\""
-       (state/config :frontend-url)
-       "/#"
-       (bidi/path-for shared-constants/routes :route.offerings/detail :offering/address offering)
-       "\""))
+(defn form-link [offering]
+  "If frontens host is not among pushroute-hosts form link with a #."
+  (let [frontend (state/config :frontend)
+        host (:host frontend)
+        port (:port frontend)
+        url (str host (when port (str ":" port)))]
+    (str "\""
+         url
+         (when-not (contains (set (state/config :pushroute-hosts)) host)
+           "/#"
```

Comment:
We don't really need to mess up with this hashroutes vs. pushstate. Real emails will never need to link to hashroute url. Please remove all code that was required for this and use only pushstate urls version.